drivers/public_reports/app/models/public_reports/homeless_population.rb

```diff
@@ -0,0 +1,576 @@
+        group(:household_id).
+        having(nf('count', [shs_t[:age].gteq(18)]).eq(0).and(nf('count', [shs_t[:age].lt(18)]).gt(1))).
+        select(:household_id)
+    end
+
+    private def total_for(scope, population)
+      count = if population == :adults_with_children
+        scope.select(:household_id).distinct.count
+      else
+        scope.select(:client_id).distinct.count
+      end
+
+      word = case population
+      when :veterans
+        'Veterans'
+      when :adults_with_children, :hoh_from_adults_with_children
+        'Households'
+      else
+        'People'
+      end
+
+      pluralize(number_with_delimiter(count), word)
+    end
+
+    private def location_chart(population)
+      {}.tap do |charts|
+        quarter_dates.each do |date|
+          case population
+          when :housed
+            charts[date.iso8601] = {
+              data: [
+                ['Rapid-Rehousing', with_service_in_quarter(report_scope, date, population).in_project_type(13).select(:client_id).distinct.count],
+                ['Permanent Housing', with_service_in_quarter(report_scope, date, population).in_project_type([3, 9, 10]).select(:client_id).distinct.count],
+              ],
+              title: _('Type of Housing'),
+              total: total_for(with_service_in_quarter(report_scope, date, population), population),
+            }
+          when :homeless
+            charts[date.iso8601] = {
+              data: [
+                ['Sheltered', with_service_in_quarter(report_scope, date, population).homeless_sheltered.select(:client_id).distinct.count],
+                ['Unsheltered', with_service_in_quarter(report_scope, date, population).homeless_unsheltered.select(:client_id).distinct.count],
+              ],
+              title: _('Where People are Staying'),
+              total: total_for(with_service_in_quarter(report_scope, date, population), population),
+            }
+          else
+            charts[date.iso8601] = {
+              data: [
+                ['Homeless', with_service_in_quarter(report_scope, date, population).homeless.select(:client_id).distinct.count],
+                ['Housed', with_service_in_quarter(report_scope, date, population).residential_non_homeless.select(:client_id).distinct.count],
+              ],
+              title: _('Homeless or Housed'),
+              total: total_for(with_service_in_quarter(report_scope, date, population), population),
+            }
+          end
+        end
+      end
+    end
+
+    private def with_service_in_quarter(scope, date, population)
+      service_scope = if population == :housed
+        GrdaWarehouse::ServiceHistoryService.where(shs_t[:date].gt(she_t[:move_in_date]))
+      else
+        :current_scope
+      end
+      scope_for(population, date, scope).with_service_between(start_date: date, end_date: date.end_of_quarter, service_scope: service_scope)
+    end
+
+    # NOTE: this count is equivalent to OutflowReport.exits_to_ph
```

Comment:
Obsolete comment?

Reply:
I don't think so, the following still uses OutflowReport.